includes/class-wp-job-manager-post-types.php

```diff
@@ -609,6 +610,16 @@ public function job_feed() {
 			];
 		}
 
+		// Hide expired positions from the job feed.
```

Comment:
Ah, I see that now. Will remove it as unneeded 👍🏻 .

Reply:
What do you think about the check added in `get_job_listings()`?

Reply:
I'm wondering if I'm following along as I should be, but here we go.

 I added the check in `get_job_listings()` as originally, the **Hide expired positions** option was _not_ hiding expired posts from the Recent Jobs widget. Or at least, that's the behaviour I was seeing at first.

When I was working on the check, I had added to the `$args` array with:

```
'featured'          => null,
'filled'            => null,
'expired'           => null,
```

And then checked with:

```
! is_null( $args['expired'] ) 
```

I had also thought about this:

```
'expired' === ($args['post_status'] )
```

But I decided that the second check wasn't great, or least as an OR, as it would simply hide `expired` posts and never make it to the option check (although that seems to be the behaviour anyway), e.g:

```
if ( 'expired' === ($args['post_status'] ) || 1 === absint( get_option( 'job_manager_hide_expired' ) ) ) {
```

In my testing, yesterday and today, however, expired posts _are_ being excluded from the Recent Jobs widget, even with the check and related `meta_query` manipulation commented out. So I don't know why I was seeing the behaviour I was originally, but I'm unable to repeat it. 

**But also**

The check doesn't work as I'd expected. If the comparison is reversed, all jobs are hidden from the Recent Jobs widget, not just those that are expired (i.e. have a past date). It seems to be excluding based simply on the existence of the `_job_expires` key, and so isn't properly taking (1) the key and doing (2) a date comparison (i.e., if the value date is less than today's date, the post is expired). What seems to be happening, is that the value is being set to today's date... but I'm unclear on that, mostly.

It seems though that removing the check is fine, as I'm not able to replicate any issues, and the Recent Jobs widget behaves as expected with the option to hide expired posts enabled.

Reply:
Sorry, I'm a little lost about which checks you're referring to here. The [post status option check](https://github.com/Automattic/wp-job-manager/blob/c2ef474d5b01ca51e1567dc7fe434870d782ee4d/wp-job-manager-functions.php#L49-L55) should catch the jobs already switched to expired (manually or by the job). 

I'm not sure what was happening with expired jobs showing up in your Recent Jobs widget. Results do get cached, so I'm not sure if that was related. 

> The check doesn't work as I'd expected. If the comparison is reversed, all jobs are hidden from the Recent Jobs widget, not just those that are expired (i.e. have a past date). It seems to be excluding based simply on the existence of the _job_expires key

Which comparison are you reversing here?

Reply:
> The [post status option check](https://github.com/Automattic/wp-job-manager/blob/c2ef474d5b01ca51e1567dc7fe434870d782ee4d/wp-job-manager-functions.php#L49-L55) should catch the jobs already switched to expired (manually or by the job).

That check works as expected for me. Jobs listings that are beyond their expiration date are correctly given an `expired` post status, or `publish` post status if the expiration date changes to a new future date. Posts are then shown, or hidden, from areas like the Recent Jobs widget based on that status alone.

> Results do get cached, so I'm not sure if that was related.

I think that's what was happening.

> Which comparison are you reversing here?

Sorry, [the comparison here](https://github.com/Automattic/WP-Job-Manager/blob/c2ef474d5b01ca51e1567dc7fe434870d782ee4d/wp-job-manager-functions.php#L143). If I switch `compare` to check for less than/equal to `date`, then _all_ Job listings are removed from the Recent Jobs widget, not just those with a `_job_expires` meta key with date value in the past.

(The `get_jobs_listing()` method is used in the Recent Jobs widget [here](https://github.com/Automattic/WP-Job-Manager/blob/28f871e3bac1bede01af6186bc4525fe38220c63/includes/widgets/class-wp-job-manager-widget-recent-jobs.php#L100), which sets the query to retrieve posts).

Reply:
> If I switch compare to check for less than/equal to date, then all Job listings are removed from the Recent Jobs widget, not just those with a _job_expires meta key with date value in the past.

These greater-than and less-than always make my brain melt when in the `meta_value` query syntax, but could it be that reversing the query is asking for job listings that expired in the past **and** still have a `publish` status (from the earlier `post_status` check that you mentioned above)?